system/jlib/jhash.hpp

```diff
@@ -49,27 +49,30 @@ interface jlib_decl IAtom : extends IMapping
 {
  public:
     virtual const char * getNamePtr() const = 0;
+    const char *         getAtomNamePtr() const { return IAtom::getAtomNamePtr(this);} // ok if this=NULL
```

Comment:
In effect you are fooling the compiler again - this is still NULL in this function, but it is no longer detecting the fact because of the call to a static member.

As discussed elsewhere, I think we need to fix properly rather than just hiding the error from the compiler.

At the same time I'd be inclined to rationalize the interface to remove the 3 different synonyms.

@ghalliday What are your thoughts?